### PR TITLE
[IMP] loading: check missing ir_model_access per module

### PR DESCRIPTION
**Pull request by Xavier-Do**
Since e1a5ed51dbb9417b357db14eff8ca22400a2e10f, missing ir_model_access are warned at the end of an install.
Unfortunately, it is still possible that a module add a model and another module add the corresponding ir_model_access.
Since runbot install all module at once, this won't be spot until a single module build is ran when each module is
installed independently.

This commit proposes to move the check at the end of each module.

Note that the log may be repeated multiple times if multiple modules redefine this model.
Note that the log will only occur on install and upgrade, not basic registry loading.

Linked to #59193

### CODE REVIEW

**Review by d-fence — APPROVED** [review]

File: odoo/modules/loading.py [hunk 1]
```diff
@@ -246,6 +246,12 @@ def load_module_graph(cr, graph, status=None, perform_checks=True,
             # update made to the schema or data so the tests can run
             # (separately in their own transaction)
             cr.commit()
+            if model_names:
+                cr.execute("""select model from ir_model where id NOT IN (select distinct model_id from ir_model_access) AND model IN %s""", (tuple(model_names),))
+                for [model] in cr.fetchall():
+                    if model in registry and not registry[model]._abstract:
```
Comment (rco-odoo) [inline odoo/modules/loading.py:252]: by construction, all `model_names` are in the registry
```suggestion
                    if not registry[model]._abstract:
```

File: odoo/modules/loading.py [hunk 2]
```diff
@@ -246,6 +246,12 @@ def load_module_graph(cr, graph, status=None, perform_checks=True,
             # update made to the schema or data so the tests can run
             # (separately in their own transaction)
             cr.commit()
+            if model_names:
+                cr.execute("""select model from ir_model where id NOT IN (select distinct model_id from ir_model_access) AND model IN %s""", (tuple(model_names),))
```
Comment (rco-odoo) [inline odoo/modules/loading.py:250]: newlines are cheap, and readability counts
```suggestion
                query = """ SELECT model FROM ir_model WHERE model IN %s
                            AND id NOT IN (SELECT DISTINCT model_id FROM ir_model_access) """
                cr.execute(query, [tuple(model_names)])
```

**Review by rco-odoo — APPROVED** [review]

File: odoo/modules/loading.py [hunk 3]
```diff
@@ -246,6 +246,19 @@ def load_module_graph(cr, graph, status=None, perform_checks=True,
             # update made to the schema or data so the tests can run
             # (separately in their own transaction)
             cr.commit()
+            regular_models = [model for model in model_names if not registry[model]._abstract]
```
Comment (rco-odoo) [inline odoo/modules/loading.py:249]: I suggest `concrete_models`, as "concrete" is the opposite of "abstract".

File: odoo/modules/loading.py [hunk 4]
```diff
@@ -246,6 +246,19 @@ def load_module_graph(cr, graph, status=None, perform_checks=True,
...
+            if regular_models:
+                cr.execute("""select model from ir_model where id NOT IN (select distinct model_id from ir_model_access) AND model IN %s""", (tuple(regular_models),))
```
Comment (rco-odoo) [inline odoo/modules/loading.py:251]: I suggest to pass parameters as a list, as it is more readable than tuples.
I also suggest to uppercase SQL keywords, and put `model IN %s` before `id IN ...`

File: odoo/modules/loading.py [hunk 5]
```diff
@@ -246,6 +246,19 @@ def load_module_graph(cr, graph, status=None, perform_checks=True,
...
+                models = [model for [model] in cr.fetchall()]
+                if models:
+                    lines = [
+                        f"The models {models} have no access rules while initiating module {module_name}, consider adding them. E.g.:"
```
Comment (rco-odoo) [inline odoo/modules/loading.py:255]:
```suggestion
                        f"The models {models} have no access rules in module {module_name}, consider adding some, like:"
```

**Comment by Xavier-Do** [comment]
robodoo r+

**Pull request closed without merging**